File: app/src/main/java/com/example/jaguar/weather/model/ParseWeather.java
package com.example.jaguar.weather.model;

import org.json.JSONObject;
import java.util.ArrayList;
import java.util.List;

public class ParseWeather {
    private List<WeatherObj> list = new ArrayList<>();

    public List<WeatherObj> parseWeather(JSONObject json){

        for(int i=0; i<6; i++) {
            try {
                String location = json.getJSONObject("location").getString("name");
                String date = json.getJSONObject("forecast").getJSONArray("forecastday").getJSONObject(i).getString("date");
                String temp = json.getJSONObject("forecast").getJSONArray("forecastday").getJSONObject(i).getJSONObject("day").getString("maxtemp_c");
                String speedwind = json.getJSONObject("forecast").getJSONArray("forecastday").getJSONObject(i).getJSONObject("day").getString("maxwind_kph");
                String humidity = json.getJSONObject("forecast").getJSONArray("forecastday").getJSONObject(i).getJSONObject("day").getString("avghumidity");
                String UrliconWeather = json.getJSONObject("forecast").getJSONArray("forecastday").getJSONObject(i).getJSONObject("day").getJSONObject("condition").getString("icon");

                list.add(new WeatherObj("https:" + UrliconWeather,date+"\n"+"Город: " + location + "\n"+
                        "Макс. температура " + temp + "\n" +
                        "Макс. скорость ветра " + speedwind + " km/h" + "\n" +
                        "Влажность " + humidity + " %\n"));
            } catch (Exception e) {
            }
        }
        return list;
    }
}

File: app/src/main/java/com/example/jaguar/weather/common/ParseWeather.kt
package com.example.jaguar.weather.common

import com.example.jaguar.weather.interfaces.CallBack
import com.example.jaguar.weather.interfaces.DaggerWeatherComponent
import com.example.jaguar.weather.interfaces.WeatherService
import retrofit2.*
import java.util.ArrayList
import javax.inject.Inject

class ParseWeather {
    @Inject lateinit var retrofit: Connection

    private val listWe = ArrayList<WeatherObj>()

    fun getWeatherFullTime(str: String, callBack: CallBack) {
        DaggerWeatherComponent.create().inject(this)

        val weatherService : WeatherService = retrofit.some().create(WeatherService::class.java)
        val weather : Call<UserModelResponse> = weatherService.getWeather(
                ConstApi.ACCSESS_TOKEN, str, "ru", "7")

        weather.enqueue(object : Callback<UserModelResponse> {
            override fun onResponse(call: Call<UserModelResponse>, response: Response<UserModelResponse>) {
                listWe.clear()
                val code = response.code()
                when (code) {
                    200 ->
                        for (i in 0..5) {
                            listWe.add(WeatherObj("Город: " + response.body()!!.location.name,
                                    response.body()!!.forecast.forecastday[i].day.condition.icon,
                                    response.body()!!.forecast.forecastday[i].day.condition.text,
                                    response.body()!!.forecast.forecastday[i].date))
                            listWe.add(WeatherObj("Город: " + response.body()!!.location.name,
                                    response.body()!!.forecast.forecastday[i].day.condition.icon,
                                    response.body()!!.forecast.forecastday[i].day.condition.text,
                                    response.body()!!.forecast.forecastday[i].date))
                            listWe.add(WeatherObj("Город: " + response.body()!!.location.name,
                                    response.body()!!.forecast.forecastday[i].day.condition.icon,
                                    response.body()!!.forecast.forecastday[i].day.condition.text,
                                    response.body()!!.forecast.forecastday[i].date))
                        }
                    400 -> println(" 404!!!!!!!!!!!")
                    else -> println(" что-то пошло не так")
                }
                callBack.UpdateWeather(listWe)
            }
            override fun onFailure(call: Call<UserModelResponse>, t: Throwable) {
                println(t)
            }
        })
    }
}